src/stage1/migrate_info.rs

```diff
@@ -202,10 +213,22 @@ impl MigrateInfo {
         &self.to_dir
     }
 
+    pub fn get_device_type_name(&self) -> String {
+         self.device.to_string()
+    }
+
     pub fn is_x86(&self) -> bool {
         self.device.supports_device_type(DEV_TYPE_GEN_X86_64)
     }
 
+    pub fn is_jetson_xavier(&self) -> bool {
+        self.device.supports_device_type(DEV_TYPE_JETSON_XAVIER)
+    }
+
+    pub fn is_jetson_xavier_nx(&self) -> bool {
+        self.device.supports_device_type(DEV_TYPE_JETSON_XAVIER_NX)
+    }
```

Comment:
Well, your code is really just doing what `is_x86()` already did. It was [my commit that introduced `--no-dt-check`](https://github.com/balena-os/takeover/pull/54/commits/91e731c3c2e333d781d50c715469da8fde683b2b) that break things. I didn't anticipate this interaction between `--no-dt-check` and the points in the code where we check for certain device types (or "device families").

I am always happy to hear suggestions, but feel free to focus on my other comments. This point here is on me!

Reply:
What where the errors on the Pi3? Was it related to what was reported in /proc/device-tree/model ?

Reply:
> Was it related to what was reported in /proc/device-tree/model ?

Yes, this. If I recall correctly, my Pi 3 reports itself as a "3 Model B Plus Rev 1", and our regex couldn't handle both "Plus" and "Rev". Or something like that.

Some more context: We decided to go with these `--no-*-check` flag instead of adjusting the regex (and similar changes for any other DTs and OSes) because this path would require continuous maintenance of `takeover` as the ecosystem evolves, and for the foreseeable future we'd rather spend this effort in other tasks.

Reply:
Wondering if we should care if it's a Pi3 B or B+, with or without a rev, because it should be able to run both the 32bit and the 64bit balenaOS images. Looking at https://www.seeedstudio.com/blog/2019/11/29/raspberry-pi-3-b-comparison-with-3b-and-more/ I see mostly networking improvements, nothing that prevent it from booting...

Reply:
@acostach , I was experimenting a bit, and came up with [this commit on a test branch](https://github.com/balena-os/takeover/commit/8ea4be5d4a4624884016e6684b86bcf94f95c629). It makes sure that no DT-specific code runs when we pass `--no-dt-check`.

I tested this commit on top of your branch and migrated my Pi 3 (from Raspberry Pi OS) passing `--no-dt-check` without problems.

What do you think? Feel free to cherry-pick, adapt or steal as you see fit!

(Rant: The whole handling of DTs on Takeover feels a bit over-complicated to me. There's so much code around the `Device` trait and all its implementations, yet they seem provide so little functionality... they don't even encapsulate the device-specific code we have :disappointed: End of rant. And this was just a rant, changing this wouldn't make sense at this point, IMO.)

Reply:
@lmbarros lgtm. Should I bring your commit in this PR?

Reply:
I think so @acostach !

Reply:
thanks @lmbarros, done!